pkg/cloudprovider/rhsm/subscription_manager.go

```diff
@@ -17,20 +17,22 @@ limitations under the License.
 package rhsm
 
 import (
+	"context"
 	"encoding/json"
 	"errors"
 	"fmt"
 	"io/ioutil"
 	"net/http"
-	"net/url"
-	"strings"
 	"time"
 
+	"golang.org/x/oauth2"
+
 	"k8s.io/klog"
 )
 
 // RedHatSubscriptionManager is responsible for removing redhat subscriptions.
 type RedHatSubscriptionManager interface {
+	//TODO(irozzo) add context in input to give more control to the caller
```

Comment:
Well adding this might interfere with the logic of the default `UnregisterInstance` method implementation, since everything in the rhsm is abstracted away from the caller(cloud provider) and we are not exposing anything.

Reply:
What I meant here is that it would be nice to provide the context explicitly by the caller of `UnregisterInstance`, so that it can set timeout and cancellation for instance.